tycho-ethereum/src/rpc/mod.rs

```diff
@@ -248,115 +284,126 @@ impl EthereumRpcClient {
         .collect())
     }
 
-    pub(crate) async fn batch_fetch_accounts_code_and_balance(
+    pub(crate) async fn fetch_accounts_code_and_balance(
         &self,
         block_id: BlockNumberOrTag,
         addresses: &[Address],
     ) -> Result<HashMap<Address, (Bytes, U256)>, RPCError> {
-        let batching = self
-            .batching
-            .as_ref()
-            .ok_or(RPCError::SetupError(
-                "BatchingConfig is required for batch operations".to_string(),
-            ))?;
+        if self.batching.supported {
+            self.batch_fetch_accounts_code_and_balance(block_id, addresses)
+                .await
+        } else {
+            self.non_batch_fetch_accounts_code_and_balance(block_id, addresses)
+                .await
+        }
+    }
+
+    async fn batch_fetch_accounts_code_and_balance(
+        &self,
+        block_id: BlockNumberOrTag,
+        addresses: &[Address],
+    ) -> Result<HashMap<Address, (Bytes, U256)>, RPCError> {
+        if !self.batching.supported {
```

Comment:
If not supported, isn't it better to fallback to sending the requests sequentially instead of raising an error?